### DISCUSSION TITLE
Installing MOOSE on windows with MSYS2

### DISCUSSION THREAD

**Discussion by starkekr** [idaholab/moose#23681] [Q&A Getting Started] [answered]
I have been able to install libmesh and petsc with msys2 but run into trouble when trying to build MOOSE. Specifically, when linking with pcre I get the following error after running make in the test folder:

Linking Executable /home/stark/projects/moose/test/moose_test-opt...
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/12.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:/msys64/home/stark/projects/moose/framework/libmoose-opt.a(mesh_Unity.x86_64-w64-mingw32.opt.o):mesh_Unity.C:(.text+0xbd0d): undefined reference to `__imp__ZN7pcrecpp2RE4InitERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEEPKNS_10RE_OptionsE'

I realize that a native windows build is not fully supported in MOOSE (outside of WSL) but nevertheless I am trying. Any thoughts would be greatly appreciated.

**Comment by GiudGiud** [comment]
I'm surprised you got this far. 

@dschwen do you know who can help

**Reply by GiudGiud** [reply, marked as answer]
So I talked to Daniel and while it s possible to get it to compile and to pass some of the test suite, the dynamic linking works differently so you wont be able to create an application from this MOOSE, which limits the usefulness quite a bit

We recommend you switch to WSL2 which has been satisfactory for many

**Reply by starkekr** [reply]
Could I just put my code into one of the modules of MOOSE and build it locally rather than using stork to build an application? I think there would be no need to build my application dynamically in that case right?

**Reply by GiudGiud** [reply]
@permcody correct me if I m wrong but I think modules are linked dynamically too

**Reply by permcody** [reply]
No, modules are compiled in, but yes, they each have their own library file.
